pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -256,22 +264,15 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
           tierToInstancePartitionsMap, null);
     }
 
-    if (currentAssignment.equals(targetAssignment)) {
+    boolean segmentAssignmentUnchanged = currentAssignment.equals(targetAssignment);
+    LOGGER.info("For rebalanceId: {},  segmentAssignmentUnchanged: {}, "
+            + "tierInstancePartitionsUnchanged: {}, instancePartitionsUnchanged: {}",
+        rebalanceJobId, segmentAssignmentUnchanged, tierInstancePartitionsUnchanged, instancePartitionsUnchanged);
+
+    if (segmentAssignmentUnchanged && tierInstancePartitionsUnchanged && instancePartitionsUnchanged) {
```

Comment:
We want to follow the existing logic, and always terminate when segment assignment is not changed:
```
if (segmentAssignmentUnchanged) {
  if (instancePartitionsUnchanged && tierInstancePartitionsUnchanged) {
    // NO_OP
    ...
  } else {
    // DONE
    if (dryRun) {
      ...
    } else {
      ...
    }
  }
}
```

Reply:
why would these be DONE if they need changing? Why not IN_PROGRESS
```
// DONE
    if (dryRun) {
      ...
    } else {
      ...
    }
```

Reply:
Why are we always returning IN_PROGRESS? https://github.com/apache/pinot/blob/master/pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java#L683-L691